utils/src/main/scala/za/co/absa/enceladus/utils/validation/SchemaValidator.scala

```diff
@@ -37,9 +38,9 @@ object SchemaValidator {
     val flatSchema = flattenSchema(schema)
     for {s <- flatSchema} {
       val fieldWithPath = if (s.structPath.isEmpty) s.field else s.field.copy(name = s.structPath + "." + s.field.name)
-      val issues = validateColumnName(s.field.name, s.structPath) ++ FieldValidator.validate(fieldWithPath)
+      val issues = validateColumnName(s.field.name, s.structPath) ++ TypedStructField(fieldWithPath).validate()
       if (issues.nonEmpty) {
-        val pattern = if (s.field.metadata contains "pattern") s.field.metadata.getString("pattern") else ""
+        val pattern = if (s.field.metadata contains "pattern") s.field.metadata.getString("pattern") else "" //TODO look into this
```

Comment:
This is the TODO mentioned in my the initial comment. Created Issue #787 about this. TODO has been removed, just not committed yet.